crates/client/src/lib.rs

```diff
@@ -33,20 +34,22 @@ pub mod storage;
 pub use self::config::*;
 
 /// A client for a Warg registry.
-pub struct Client<P, C> {
+pub struct Client<P, C, T> {
     packages: P,
     content: C,
     api: api::Client,
+    checkpoint: T,
 }
 
-impl<P: PackageStorage, C: ContentStorage> Client<P, C> {
+impl<P: PackageStorage, C: ContentStorage, T: CheckpointStorage> Client<P, C, T> {
```

Comment:
I think we should probably rename `PackageStorage` to `RegistryStorage` and incorporate the checkpoint into it; eventually we'll need to add in operator log reading/writing into it and I'd rather not have more traits.

The reason there's two traits here is that content storage is across all registries, whereas the rest of the data is per-registry.